tutorials/cifar10-distributed.ipynb

```diff
@@ -0,0 +1,1153 @@
+        "\n",
+        "        with autocast(enabled=with_amp):\n",
+        "            y_pred = model(x)\n",
+        "            loss = criterion(y_pred, y)\n",
+        "\n",
+        "        optimizer.zero_grad()\n",
+        "        scaler.scale(loss).backward()  # If with_amp=False, this is equivalent to loss.backward()\n",
+        "        scaler.step(optimizer)  # If with_amp=False, this is equivalent to optimizer.step()\n",
+        "        scaler.update()  # If with_amp=False, this step does nothing\n",
+        "\n",
+        "        return {\n",
+        "            \"batch loss\": loss.item(),\n",
+        "        }\n",
+        "\n",
+        "    trainer = Engine(train_step)\n",
+        "    trainer.logger = logger\n",
+        "\n",
+        "    to_save = {\n",
+        "        \"trainer\": trainer,\n",
+        "        \"model\": model,\n",
+        "        \"optimizer\": optimizer,\n",
+        "        \"lr_scheduler\": lr_scheduler,\n",
+        "    }\n",
+        "    metric_names = [\n",
+        "        \"batch loss\",\n",
+        "    ]\n",
+        "\n",
+        "    common.setup_common_training_handlers(\n",
+        "        trainer=trainer,\n",
+        "        train_sampler=train_sampler,\n",
+        "        to_save=to_save,\n",
+        "        save_every_iters=config[\"checkpoint_every\"],\n",
+        "        save_handler=get_save_handler(config),\n",
+        "        lr_scheduler=lr_scheduler,\n",
+        "        output_names=metric_names if config[\"log_every_iters\"] > 0 else None,\n",
+        "        with_pbars=False,\n",
+        "        clear_cuda_cache=False,\n",
+        "    )\n",
+        "\n",
+        "    if config[\"resume_from\"] is not None:\n",
+        "        checkpoint = load_checkpoint(config[\"resume_from\"])\n",
+        "        Checkpoint.load_objects(to_load=to_save, checkpoint=checkpoint)\n",
+        "\n",
+        "    return trainer"
+      ],
+      "execution_count": 54,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "U5ius6EM9aiG"
+      },
+      "source": [
+        "## Evaluator\n",
+        "\n",
+        "The evaluator will be created from `evaluate_step` which will:\n",
+        "1. Set the `model` to `eval()` mode.\n",
+        "2. Move the batch to `device` used in current distributed configuration.\n",
+        "3. Perform forward pass. If AMP is enabled, `autocast` will be on.\n",
+        "\n",
+        "Finally, we will attach relevant Ignite metrics to the `evaluator`. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "JPVzU6CqNlE4"
+      },
+      "source": [
+        "def create_evaluator(model, metrics, config):\n",
+        "    with_amp = config[\"with_amp\"]\n",
+        "    device = idist.device()\n",
+        "\n",
+        "    @torch.no_grad()\n",
+        "    def evaluate_step(engine, batch):\n",
```

Comment:
Same for the evaluation

https://github.com/pytorch/ignite/blob/31738d47716d3055c356f29b9e01e022062041ea/ignite/engine/__init__.py#L518